app/src/main/java/com/android/calendar/widget/CalendarAppWidgetModel.java

```diff
@@ -111,18 +128,34 @@ public void buildFromCursor(Cursor cursor, String timeZone) {
             int i = mEventInfos.size();
             mEventInfos.add(populateEventInfo(eventId, allDay, start, end, startDay, endDay, title,
                     location, color, selfStatus));
+        }
+    }
+
+    public void populateBuckets(String timeZone) {
+        final Time recycle = new Time(timeZone);
+        recycle.set(System.currentTimeMillis());
+        Collections.sort(mEventInfos, Comparator.comparing(u -> new Date(u.start)));
+
+        final ArrayList<LinkedList<RowInfo>> mBuckets =
+                new ArrayList<LinkedList<RowInfo>>(CalendarAppWidgetService.MAX_DAYS);
+        for (int i = 0; i < CalendarAppWidgetService.MAX_DAYS; i++) {
+            mBuckets.add(new LinkedList<RowInfo>());
+        }
+        int i = 0;
+        for (EventInfo mEventInfo : mEventInfos) {
             // populate the day buckets that this event falls into
-            int from = Math.max(startDay, mTodayJulianDay);
-            int to = Math.min(endDay, mMaxJulianDay);
+            int from = Math.max(Time.getJulianDay(mEventInfo.start, new Time().getGmtOffset()), mTodayJulianDay);
```

Comment:
can you give some explanation here?

Reply:
same as before. tasks has no begin time, so it will be recalcualted

Reply:
Would it be possible to compute them in the object instead of different times in the code?

Reply:
@mebitek just a friendly pint. Do you have any comment here? I think I would be fine with the current state but maybe this would simplify the code a bit.